SimTKcommon/src/ParallelExecutor.cpp

```diff
@@ -25,23 +25,36 @@
 #include "SimTKcommon/internal/ParallelExecutor.h"
 #include <pthread.h>
 
+#include <iostream>
+#include <string>
+#include <algorithm>
+
+using namespace std;
+
 namespace SimTK {
 
 static void* threadBody(void* args);
 
+ParallelExecutorImpl::ParallelExecutorImpl() : finished(false) {
+
+    // Set the maximum number of threads that we can use
+    SimTK_APIARGCHECK_ALWAYS(ParallelExecutor::getNumProcessors() > 0, "ParallelExecutorImpl", "ParallelExecutorImpl", "Number of threads detected is not positive.");
+    NUM_MAX_THREADS = ParallelExecutor::getNumProcessors();
+
+    pthread_mutex_init(&runLock, NULL);
```

Comment:
This now has duplicate code in both constructors -- should only appear once. You can invoke one constructor from the other one, or split the code out into a private `init()` method or something like that.